### PR TITLE
Makefile

### PR DESCRIPTION
**Pull request by Bslabe123**
Added a makefile to cover a few common use cases:
- `update-deps`: generate the requirements.txt and install dependencies from it
- `generate-protos:`: update all protos the repo
- `check`: run the `pylint` linter
- `format`: format code

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+generate-protos: generate-and-prepend-preambles format
+
+format:
+	$(PIP) install pyink
+	pyink --pyink-indentation 2 --line-length 80 --verbose .
+
+check:
+	pylint --ignore-patterns=".*_pb2.py,.*_pb2_grpc.py" jetstream/ benchmarks/
+
```
Comment (FanhaiLu1) [inline Makefile:21]: Thanks for adding this one! What is the benefit of it compared with current approach? It seems more complex.

File: .github/workflows/unit_tests.yaml [hunk 2]
```diff
@@ -53,7 +53,7 @@ jobs:
         pytype --jobs auto --disable=import-error,module-attr jetstream/ benchmarks/
     - name: Analysing the code with pylint
       run: |
-        pylint jetstream/ benchmarks/
+        make check
```
Comment (JoeZijunZhou) [inline .github/workflows/unit_tests.yaml:56]: Awesome! Do you mind also replacing other commands in this CI? Thanks!

Comment (Bslabe123) [reply .github/workflows/unit_tests.yaml:56]: Sure!

File: Makefile (hunk 1 above)
Comment (Bslabe123) [reply Makefile:21]: For adding the --ignore-patterns or for adding the target entirely? Added the --ignore-patterns flag so we dont need to add `pylint: disable=all` to files when we regenerate protos. Added the target entirely to make is usable elsewhere, for example if we want to add a precommit hook

File: .github/workflows/unit_tests.yaml (hunk 2 above)
Comment (Bslabe123) [reply .github/workflows/unit_tests.yaml]: Done

**Review by JoeZijunZhou — APPROVED** [review]
Have we formatted the Makefile?

**Comment by Bslabe123** [comment]
> Have we formatted the Makefile?

No standard way to format a makefile as far as I can see

**Pull request merged**